pallets/auctions/src/types.rs

```diff
@@ -0,0 +1,243 @@
+where
+    AccountId: Clone + Default,
+    BalanceCaps: Clone + Default,
+{
+    pub creator: AccountId,
+    pub start_block: BlockNumber,
+    pub end_block: BlockNumber,
+    pub start_price: BalanceCaps,
+    pub buy_it_price: Option<BalanceCaps>,
+    pub bidders: SortedBidderList<AccountId, BalanceCaps>,
+    pub marketplace_id: MarketplaceId,
+}
+
+#[derive(Encode, Decode, Clone, PartialEq, RuntimeDebug, TypeInfo)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+/// wrapper type to store sorted list of all bids
+/// The wrapper exists to ensure a queue implementation of sorted bids
+pub struct SortedBidderList<AccountId, BalanceCaps>(pub Vec<(AccountId, BalanceCaps)>);
+
+impl<AccountId, BalanceCaps> SortedBidderList<AccountId, BalanceCaps>
+where
+    AccountId: std::cmp::Ord + Clone,
+    BalanceCaps: std::cmp::PartialOrd,
+{
+    pub const MAX_COUNT: usize = 10;
+
+    /// Create a new empty bidders list
+    pub fn new() -> Self {
+        Self(Vec::new())
+    }
+
+    /// Insert a new bid to the list
+    ///
+    /// The function ensures that the new bid is always higher than the current highest bid
+    /// If the insert action will cause an overflow, the lowest bid is removed and returned
+    /// If value lower than current highest bid is passed, the function will panic
+    pub fn insert_new_bid(
+        &mut self,
+        account_id: AccountId,
+        value: BalanceCaps,
+    ) -> Option<(AccountId, BalanceCaps)> {
+        // ensure the new bid is larger than current highest bid
+        if let Some(current_highest_bid) = self.get_current_highest_bid() {
+            if current_highest_bid.1 >= value {
+                // this panic should never happen since the extrinsic already checks if value > current_highest_bid
+                panic!("cannot accept a lower bid!");
+            }
+        }
+        // If list is at max capacity, remove lowest bid
+        match self.0.len() {
+            Self::MAX_COUNT => {
+                let removed_bid = self.0.remove(0);
+                self.0.push((account_id, value));
+                // return removed bid
+                Some(removed_bid)
+            }
+            _ => {
+                self.0.push((account_id, value));
+                None
+            }
+        }
+    }
+
+    /// Get length of bidders list
+    pub fn len(&self) -> usize {
+        self.0.len()
+    }
+
+    /// Get current highest bid in list
+    pub fn get_current_highest_bid(&self) -> Option<&(AccountId, BalanceCaps)> {
+        self.0.last()
+    }
+
+    /// Get current lowest bid in list
+    pub fn get_current_lowest_bid(&self) -> Option<&(AccountId, BalanceCaps)> {
+        self.0.first()
+    }
+
+    /// Remove the lowest bid in list
+    pub fn remove_lowest_bid(&mut self) -> (AccountId, BalanceCaps) {
+        self.0.remove(0)
+    }
+
+    /// Remove a specific bid from list
+    pub fn remove_bid(&mut self, account_id: &AccountId) {
```

Comment:
Instead of returning nothing, return an boolean which will tell us if we have remove an entry or not.